Symfony/CS/Fixer/Contrib/DirConstantFixer.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Vladimir Reznichenko <kalessil@gmail.com>
+ */
+class DirConstantFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        // generate a new token, otherwise transformations are not possible
+        $dirTokenContainer = Tokens::fromCode('<?php __DIR__; ');
+        $dirConstantToken = $dirTokenContainer->findGivenKind(T_DIR);
+        $dirConstantToken = $dirConstantToken[1];
```

Comment:
How? Raw tokens consumed from `token_get_all`, and ids verified. There is not any token_* function to create a toke on demand, or?